midend/validateProperties.h

```diff
@@ -22,7 +22,12 @@ limitations under the License.
 
 namespace P4 {
 
-// Checks to see if there are any unknown properties.
+/**
+ * Checks to see if there are any unknown properties.
+ *
+ * @pre none
+ * @post no invalid table properties in P4 program.
```

Comment:
Actually the properties are not removed, but an error is given if such properties are found.